rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3132,10 +3279,27 @@ extern "C" rmw_ret_t rmw_send_response(
   RET_NULL(request_header);
   RET_NULL(ros_response);
   CddsService * info = static_cast<CddsService *>(service->data);
-  cdds_request_header_t header;
-  memcpy(&header.guid, request_header->writer_guid, sizeof(header.guid));
-  header.seq = request_header->sequence_number;
-  return rmw_send_response_request(&info->service, header, ros_response);
+  // Block until the response reader has been matched by the response writer (this is a
+  // workaround: rmw_service_server_is_available should keep returning false until this
+  // is a given).
+  client_present_t st;
+  while ((st =
+    check_for_response_reader(info->service, *request_header)) == client_present_t::MAYBE)
+  {
+    dds_sleepfor(DDS_MSECS(10));
+  }
+  switch (st) {
+    case client_present_t::ERROR:
+      // break instead of returns makes gcc happy
+      break;
+    case client_present_t::MAYBE:
+      return RMW_RET_TIMEOUT;
```

Comment:
@eboasson how could `st` ever be `client_present_t::MAYBE` at this point if that wouldn't allow it to leave the loop in the first place?

Reply:
True, and it would be better to combine it with the `ERROR` case. I usually go out of my way to treat `enum`s as enumerated types and cover all the cases in switches over them. But C & C++ semantics don't guarantee that one can assign only defined values to an object of an enumerated type, and what then happens is compiler dependent.

Clang doesn't warn if you cover all cases of an enum, presumably on the assumption that you treat the enum as an enum (or perhaps it proves that no other values ever get assigned to it). Gcc warns, presumably because technically other values are legal too and, again presumably, it doesn't do the analysis to prove that no other values ever get assigned to it. As the comment suggests, there used to be a return in line 3294. With that, the `MAYBE` case makes a bit more sense. Now, it just looks silly.